src/api/AzureResourceGroupsExtensionApi.ts

```diff
@@ -0,0 +1,24 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtTreeDataProvider, AzExtTreeItem } from '@microsoft/vscode-azext-utils';
+import { Disposable, TreeView } from 'vscode';
+import { AppResourceResolver, AzureResourceGroupsExtensionApi } from '../api';
+
+export class InternalAzureResourceGroupsExtensionApi implements AzureResourceGroupsExtensionApi {
+    readonly tree: AzExtTreeDataProvider;
```

Comment:
Isn't `readonly` on a class just TypeScript sugar? JavaScript has `#private` fields that we can wrap with `get`'ers only. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Reply:
Oh, yeah you're right.  TypeScript will yell at you for trying to set it, but I'll do this implementation instead.

Reply:
I think the way I did it makes sense?

The interface in the api will still just have `readonly` properties.  That's what is being implemented and when we create the api, we are only enforcing those properties are passed in.  

Those are used to create the actual ExtensionApi which uses private variables that shouldn't be able to be overwritten, but since it's private, I don't think end-users would need to know about that implementation detail, which is why just leaving it as `readonly` makes sense in the api, right?